### PR TITLE
Maplibre fix place labels

### PR DESCRIPTION
**Pull request by binh-dam-ibigroup**
Now requires #454.

This PR improves label placement and cleans up the code for origin/destination, transit and rental leg places. What leg extremities get rendered (label and place circle) should be the same as what is at https://opentripplanner.org/otp-ui, with the addition of the labels for the origin and destination of an itinerary.

### CODE REVIEW

File: packages/transitive-overlay/src/index.tsx [hunk 1]
```diff
@@ -126,8 +179,34 @@ const TransitiveCanvasOverlay = ({
...
+        layout={defaultTextLayoutParams}
+        paint={defaultTextPaintParams}
+        type="symbol"
+      />
+      <Layer
+        filter={accessLegFilter}
+        id="access-leg-circles"
+        paint={{
+          "circle-color": ["get", "color"],
+          "circle-radius": 6,
```
Comment (miles-grant-ibigroup) [inline packages/transitive-overlay/src/index.tsx:197]: Any way we can make these blobs a bit bigger? To match `itinerary-body`

File: packages/transitive-overlay/src/index.tsx [hunk 2]
```diff
@@ -176,20 +254,21 @@ const TransitiveCanvasOverlay = ({
...
+          "circle-radius": 5,
+          "circle-stroke-width": 2
         }}
-        paint={{ "text-translate": [0, 5] }}
+        type="circle"
+      />
+      <Layer
+        filter={stopFilter}
+        id="stops-labels"
+        layout={defaultTextLayoutParams}
```
Comment (miles-grant-ibigroup) [inline packages/transitive-overlay/src/index.tsx:270]: It's a bit distracting to see the station labels at the same size as the origin/destination labels. Is there any way to adjust the priority/min zoom level/font size/font color of these labels so they don't overpower the origin/destination labels?

File: packages/transitive-overlay/src/util.ts [hunk 3]
```diff
@@ -25,7 +125,7 @@ export function itineraryToTransitive(
     intl?: IntlShape;
   }
 ): TransitiveData {
-  const { isAccessMode, isFlex, isTransit } = coreUtils.itinerary;
+  console.log(itin);
```
Comment (miles-grant-ibigroup) [inline packages/transitive-overlay/src/util.ts:128]: I think this was left in by mistake!

File: packages/transitive-overlay/src/util.ts [hunk 4]
```diff
@@ -264,12 +356,13 @@ export function itineraryToTransitive(
...
   tdata.routes.push(...Object.values(routes));
-  tdata.stops.push(...Object.values(stops));
+  tdata.stops = newStops;
 
   // add the journey to the tdata journeys array
   tdata.journeys.push(journey);
+  tdata.places = newPlaces;
 
-  // console.log('derived tdata', tdata);
+  console.log("derived tdata", tdata);
```
Comment (miles-grant-ibigroup) [inline packages/transitive-overlay/src/util.ts:365]: we should comment this out again!

**Review by miles-grant-ibigroup — CHANGES_REQUESTED** [review]
Only cosmetic changes for now -- I really love the new text style, but I think there are too many labels right now to clearly tell what's going on. I think either using font-size or font-color to differentiate will probably fix.

File: packages/transitive-overlay/src/index.tsx (hunk 1 above)
Comment (binh-dam-ibigroup) [reply packages/transitive-overlay/src/index.tsx:197]: Is this comment geared towards the circles that denote transit stops?

Comment (miles-grant-ibigroup) [reply packages/transitive-overlay/src/index.tsx:197]: Yes! The bikeshare/scootershare ones too

File: packages/transitive-overlay/src/util.ts (hunk 4 above)
Comment (binh-dam-ibigroup) [reply packages/transitive-overlay/src/util.ts]: Yes, thanks, reverted in https://github.com/opentripplanner/otp-ui/pull/452/commits/7ca81c36b3491b584b3136e005ab964e017e0a30.

File: packages/transitive-overlay/src/util.ts (hunk 3 above)
Comment (binh-dam-ibigroup) [reply packages/transitive-overlay/src/util.ts]: Yes, reverted in https://github.com/opentripplanner/otp-ui/pull/452/commits/7ca81c36b3491b584b3136e005ab964e017e0a30.

File: packages/transitive-overlay/src/index.tsx (hunk 1 above)
Comment (binh-dam-ibigroup) [reply packages/transitive-overlay/src/index.tsx]: Handled in https://github.com/opentripplanner/otp-ui/pull/452/commits/dcd638db37dca2e2cf56f699818b6f2afb9fc37a.

File: packages/transitive-overlay/src/index.tsx (hunk 2 above)
Comment (binh-dam-ibigroup) [reply packages/transitive-overlay/src/index.tsx]: I was looking to some way to make the origin/destination bold, check it out in https://github.com/opentripplanner/otp-ui/pull/452/commits/8a5358d36319eb6f4b206b421e386386662748b4. With this, origin/destination will have the top priority, followed by transit stations then access leg places.

**Review by miles-grant-ibigroup — APPROVED** [review]
I really love the red tint for the arrival label! 

One small comment that might clean things up, but if I missed something then feel free to ignore.

Works great after merging in required PRs! Looks like the snapshots will be have to be updated -- the coordinates are being clipped for some reason

File: packages/transitive-overlay/src/util.ts [hunk 5]
```diff
@@ -2,12 +2,118 @@ import {
...
+  knownStopNames: Record<string, string>
+): TransitiveStop {
+  // Collapse case and spaces for comparison.
+  // ("Midtown Station" and "Midtown   STATION" are considered the same name.)
+  const normalizedStopName = stop.name.toUpperCase().replace(/\s+/g, "");
+  const stopNameExists = knownStopNames[normalizedStopName];
+  if (!stopNameExists) knownStopNames[normalizedStopName] = normalizedStopName;
+  return {
+    stop_id: stop.stopId,
+    stop_name: stopNameExists ? null : stop.name,
```
[the remark this answers is not in the document]
Comment (binh-dam-ibigroup) [reply packages/transitive-overlay/src/util.ts]: Sure, why not. Updated in https://github.com/opentripplanner/otp-ui/pull/452/commits/0f45ae55bc8ca1c5e22b7a5959629370c692f5ca.

**Comment by binh-dam-ibigroup** [comment]
@miles-grant-ibigroup @demory Per discussion, I have reinstated intermediate stops in the transitive dictionary and adjusted the logic for displaying transit stops in https://github.com/opentripplanner/otp-ui/pull/452/commits/e9b5b31092b7ae614685cc2fe152574b38a2e073.

**Review by demory — APPROVED** [review]

**Pull request merged**